pandas/core/base.py

```diff
@@ -63,7 +65,7 @@ def __repr__(self) -> str:
         # Should be overwritten by base classes
         return object.__repr__(self)
 
-    def _reset_cache(self, key=None):
+    def _reset_cache(self, key: Optional[str] = None) -> None:
```

Comment:
do we ever actually pass a key?

Reply:
Twice, both times in groupby/groupby.py.